app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,35 @@
+class CommentsController < ApplicationController
+  def index; end
+
+  def show; end
+
+  def new; end
+
+  def create
+    @post = Post.find_by id: params[:id]
```

Comment:
please create a method:
```
def find_post
  @post = Post.find_by id: params[:id]
  return if @post.present?

  flash ...
  redirect ...
end
```
Then call 'before_action' for action 'create' and 'delete'